examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart

```diff
@@ -126,7 +126,7 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
           builder: (BuildContext context) {
             return _buildBottomPicker(
               CupertinoTimerPicker(
-                initialTimerDuration: timer,
+                controller: CupertinoTimerPickerController(desiredTimerDuration: timer),
```

Comment:
You don't want to change the controller instance all the time. Also see EditableTextState. You want to change your didUpdateWidget and make sure that when you change controller instance, you swap the listeners and also update the value to the new controller's value. They should never diverge.